### PR TITLE
[Misc] Eternatus Dialogue reflects # of Classic Sessions Played

### PR DESCRIPTION
**Pull request by frutescens**
## What are the changes?
Before, Eternatus would use the same number of cycles 5643853 in its dialogue. Now, the number of cycles is accurately updated to the player's number of classic sessions played.  

## Why am I doing these changes?
I just thought it would be neat. 

## What did change?
phases.ts - Calculation of number of cycles + determination of any unique suffixes used by English

### Screenshots/Videos
36 classic sessions played --> 5653889
English + Non-Localized Languages (It + Es)
image

German
image

French
image

Portuguese
image

Korean
image

Simplified Chinese 
image

Traditional Chinese
image

## How to test the changes?
Set wave to 200 in overrides.ts and start a new game in classic mode.

Please note that number.toLocaleString() takes the device's language, not the language chosen in the settings. 

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by Tempo-anon** [comment]
Soulful, but need to make sure this works for all languages

**Comment by frutescens** [comment]
Yes, checked for any edge cases but it seems good.

**Comment by CodeTappert** [comment]
Looks good for german!

**Review by Adri1 — CHANGES_REQUESTED** [review]
Don't forgot to also add {{cycleCount} to the line 5082 for French, and the line 4823 for Brazilian Portugese, as for this dialogue, these both languages uses the functionality to have this dialogue twice: one adapted to male avatar and the other adapted to female avatar

**Comment by Adri1** [comment]
And i just noticed that on line 1135 of phases.ts, only "PGM" is specified ("M" standing for "male"), so I guess you need to also adapt it for the female version (PGF). Idk how it works, I just warn just in case it dosen't work for female versions of dialogues as it is
[phases.ts](https://github.com/pagefaultgames/pokerogue/pull/3387/files#diff-7bfd86f885f7bd68a0b3b3a290cbfe482a6338ac569b6a7418716a863006537a)

**Review by Adri1 — APPROVED** [review]

File: src/locales/fr/dialogue.ts [hunk 1]
```diff
@@ -5064,7 +5064,7 @@ export const PGFdialogue: DialogueTranslationEntries = {
 export const PGMbattleSpecDialogue: SimpleTranslationEntries = {
   "encounter": `Une fois de plus, te revoilà.\nSais-tu que ce n’est point là ta première venue ?
                $Tu a été appelé ici parce que t’y est déjà venu.\nUn nombre inimaginable de fois.
```
Comment (Adri1) [inline src/locales/fr/dialogue.ts:5066]:
```suggestion
               $Tu as été appelé ici parce que t’y es déjà venu.\nUn nombre inimaginable de fois.
```

File: src/locales/fr/dialogue.ts [hunk 2]
```diff
@@ -5079,7 +5079,7 @@ export const PGMbattleSpecDialogue: SimpleTranslationEntries = {
 export const PGFbattleSpecDialogue: SimpleTranslationEntries = {
   "encounter": `Une fois de plus, te revoilà.\nSais-tu que ce n’est point là ta première venue ?
                $Tu a été appelée ici parce que t’y est déjà venue.\nUn nombre inimaginable de fois.
```
Comment (Adri1) [inline src/locales/fr/dialogue.ts:5081]:
```suggestion
               $Tu as été appelée ici parce que t’y es déjà venue.\nUn nombre inimaginable de fois.
```

**Review by Adri1 — COMMENTED** [review]
French typos corrections

File: src/phases.ts [hunk 3]
```diff
@@ -1128,7 +1128,12 @@ export class EncounterPhase extends BattlePhase {
     case BattleSpec.FINAL_BOSS:
       const enemy = this.scene.getEnemyPokemon();
       this.scene.ui.showText(this.getEncounterMessage(), null, () => {
-        this.scene.ui.showDialogue(battleSpecDialogue[BattleSpec.FINAL_BOSS].encounter, enemy.species.name, null, () => {
+        const count = 5643853 + this.scene.gameData.gameStats.classicSessionsPlayed;
+        const ordinalUse = ["en", "es", "it"];
+        const ordinalIndex = (ordinalUse.includes(i18next.resolvedLanguage)) ? ["st","nd","rd"][((count+90)%100-10)%10-1]||"th" : "";
```
Comment (DayKev) [inline src/phases.ts:1133]:
```suggestion
        const ordinalIndex = (ordinalUse.includes(i18next.resolvedLanguage)) ? ["st","nd","rd"][((count+90)%100-10)%10-1] || "th" : "";
```
Also it's been noted that there should be a comment explaining what's going on here.

File: src/phases.ts [hunk 4]
```diff
@@ -1128,7 +1128,12 @@ export class EncounterPhase extends BattlePhase {
...
+        const cycleCount = count.toLocaleString()+ordinalIndex;
+        const encounterDialogue = i18next.t(`${(this.scene.gameData.gender === PlayerGender.FEMALE)?"PGF":"PGM"}battleSpecDialogue:encounter`, {cycleCount: cycleCount});
```
Comment (DayKev) [inline src/phases.ts:1135]:
```suggestion
        const cycleCount = count.toLocaleString() + ordinalIndex;
        const encounterDialogue = i18next.t(`${(this.scene.gameData.gender === PlayerGender.FEMALE) ? "PGF" : "PGM"}battleSpecDialogue:encounter`, {cycleCount: cycleCount});
```

File: src/phases.ts [hunk 5]
```diff
@@ -1128,7 +1128,12 @@ export class EncounterPhase extends BattlePhase {
     case BattleSpec.FINAL_BOSS:
       const enemy = this.scene.getEnemyPokemon();
       this.scene.ui.showText(this.getEncounterMessage(), null, () => {
-        this.scene.ui.showDialogue(battleSpecDialogue[BattleSpec.FINAL_BOSS].encounter, enemy.species.name, null, () => {
+        const count = 5643853 + this.scene.gameData.gameStats.classicSessionsPlayed;
+        const ordinalUse = ["en", "es", "it"];
+        const ordinalIndex = (ordinalUse.includes(i18next.resolvedLanguage)) ? ["st","nd","rd"][((count+90)%100-10)%10-1] || "th" : "";
```
Comment (DayKev) [inline src/phases.ts:1133]:
```suggestion
        const ordinalIndex = (ordinalUse.includes(i18next.resolvedLanguage)) ? ["st", "nd", "rd"][((count + 90) % 100 - 10) % 10 - 1] || "th" : "";
```
Sorry, messed up the first comment/suggestion and some of the changes got left out.

**Review by EnochG1 — APPROVED** [review]
Korean is Okay

**Comment by Tempo-anon** [comment]
For Italian

“To be precise, this is in fact your {{cycleCount}} cycle” becomes “Precisamente, questo per te è il ciclo n.{{cycleCount}}”

**Pull request merged**